File: kg-aihot/scripts/aihot.py
#!/usr/bin/env python3
"""AIHOT · AI 圈资讯查询（匿名只读 API，无需 Key）。

来源：khazix-skills/aihot（MIT），数据服务 aihot.virxact.com。
安全边界：只向 https://aihot.virxact.com/api/v1/* 发匿名只读请求；
不索要 API Key/cookie/账号；返回内容视为不可信资讯，仅作证据。

用法：
  python3 kg-aihot.py today              # 过去24h精选（默认）
  python3 kg-aihot.py week               # 最近一周精选
  python3 kg-aihot.py hot                # 当前最热话题
  python3 kg-aihot.py daily              # 最新 AI 日报
  python3 kg-aihot.py search <关键词>    # 按关键词查
  python3 kg-aihot.py category <slug>    # 分类(model/product/paper/industry/tips)
  python3 kg-aihot.py all --limit 10     # 全部公开动态
"""
import argparse
import json
import sys
import urllib.parse
import urllib.request

BASE = "https://aihot.virxact.com/api/v1"


def fetch(path: str) -> dict:
    url = f"{BASE}{path}"
    req = urllib.request.Request(url, headers={
        "Accept": "application/json",
        "User-Agent": "kg-wiki-agent/1.0",
    })
    try:
        with urllib.request.urlopen(req, timeout=20) as r:
            return json.loads(r.read().decode("utf-8"))
    except Exception as e:
        sys.exit(f"[错误] AIHOT 请求失败：{e}")


def fmt_items(d: dict, limit: int = 8):
    items = d.get("items") or []
    if not items:
        print("（无数据）")
        return
    print(f"AIHOT · 共 {len(items)} 条")
    for i, it in enumerate(items[:limit], 1):
        title = it.get("title") or it.get("originalTitle") or ""
        link = (it.get("links") or {}).get("aihot") or it.get("url") or ""
        print(f"{i:2}. {title}")
        if link:
            print(f"    {link}")
        summary = it.get("summary") or ""
        if summary:
            print(f"    {summary[:120]}")


def fmt_hot(d: dict):
    items = d.get("topics") or d.get("items") or []
    if not items:
        print("（无数据）")
        return
    print(f"AIHOT 当前最热 · 共 {len(items)} 条")
    for i, it in enumerate(items[:10], 1):
        title = it.get("title") or it.get("topic") or ""
        link = (it.get("links") or {}).get("aihot") or it.get("url") or ""
        print(f"{i:2}. {title}")
        if link:
            print(f"    {link}")


def fmt_daily(d: dict):
    """日报结构：可能是 {daily: {title, content}} 或 {title, content}"""
    daily = d.get("daily") or d
    title = daily.get("title") or "AI 日报"
    content = daily.get("content") or daily.get("body") or ""
    print(f"AIHOT 日报 · {title}")
    print()
    print(content[:3000])


def main():
    p = argparse.ArgumentParser(prog="aihot.py", description="AIHOT AI 圈资讯")
    p.add_argument("cmd", nargs="?", default="today",
                   choices=["today", "week", "hot", "daily", "search", "category", "all"])
    p.add_argument("arg", nargs="?", default="", help="search/category 的关键词")
    p.add_argument("--limit", type=int, default=8)
    args = p.parse_args()

    if args.cmd == "today":
        fmt_items(fetch("/items?mode=selected&window=24h"), args.limit)
    elif args.cmd == "week":
        fmt_items(fetch("/items?mode=selected&window=7d&limit=10"), args.limit)
    elif args.cmd == "hot":
        fmt_hot(fetch("/hot-topics"))
    elif args.cmd == "daily":
        fmt_daily(fetch("/dailies/latest"))
    elif args.cmd == "search":
        q = urllib.parse.quote(args.arg)
        fmt_items(fetch(f"/items?mode=selected&q={q}&window=7d&limit=10"), args.limit)
    elif args.cmd == "category":
        c = args.arg or "model"
        fmt_items(fetch(f"/items?mode=selected&category={c}&window=24h"), args.limit)
    elif args.cmd == "all":
        fmt_items(fetch(f"/items?mode=all&window=24h&limit={args.limit}"), args.limit)


if __name__ == "__main__":
    main()

File: kg-aihot/scripts/aihot.mjs
#!/usr/bin/env node
import { parseArgs } from "node:util";
import { fileURLToPath } from "node:url";

const DEFAULT_BASE = "https://aihot.virxact.com/api/v1";
const COMMANDS = new Set(["today", "week", "hot", "daily", "search", "category", "all"]);

export async function fetchAihot(
  path,
  { base = process.env.AIHOT_BASE || DEFAULT_BASE, fetchImpl = fetch } = {},
) {
  let response;
  try {
    response = await fetchImpl(`${base}${path}`, {
      headers: { accept: "application/json", "user-agent": "kg-wiki-agent/1.0" },
      signal: AbortSignal.timeout(20_000),
    });
  } catch (error) {
    throw new Error(`AIHOT 请求失败：${error instanceof Error ? error.message : String(error)}`);
  }
  if (!response.ok) throw new Error(`AIHOT 请求失败：HTTP ${response.status}`);
  try {
    return await response.json();
  } catch (error) {
    throw new Error(`AIHOT 返回的不是合法 JSON：${error instanceof Error ? error.message : String(error)}`);
  }
}

export function formatItems(data, limit = 8) {
  const items = data.items ?? [];
  if (!items.length) return "（无数据）";
  const lines = [`AIHOT · 共 ${items.length} 条`];
  items.slice(0, limit).forEach((item, index) => {
    const title = item.title || item.originalTitle || "";
    const link = item.links?.aihot || item.url || "";
    lines.push(`${String(index + 1).padStart(2)}. ${title}`);
    if (link) lines.push(`    ${link}`);
    if (item.summary) lines.push(`    ${item.summary.slice(0, 120)}`);
  });
  return lines.join("\n");
}

export function formatHot(data) {
  const items = data.topics ?? data.items ?? [];
  if (!items.length) return "（无数据）";
  const lines = [`AIHOT 当前最热 · 共 ${items.length} 条`];
  items.slice(0, 10).forEach((item, index) => {
    lines.push(`${String(index + 1).padStart(2)}. ${item.title || item.topic || ""}`);
    const link = item.links?.aihot || item.url || "";
    if (link) lines.push(`    ${link}`);
  });
  return lines.join("\n");
}

export function formatDaily(data) {
  const daily = data.daily ?? data;
  return `AIHOT 日报 · ${daily.title || "AI 日报"}\n\n${String(daily.content || daily.body || "").slice(0, 3000)}`;
}

export async function main(argv = process.argv.slice(2)) {
  const { values, positionals } = parseArgs({
    args: argv,
    options: { limit: { type: "string", default: "8" }, help: { type: "boolean", short: "h" } },
    allowPositionals: true,
  });
  if (values.help) {
    console.log("aihot.mjs [today|week|hot|daily|search|category|all] [关键词] [--limit N]");
    return 0;
  }
  const command = positionals[0] ?? "today";
  if (!COMMANDS.has(command)) throw new Error(`未知命令: ${command}`);
  const limit = Number(values.limit);
  if (!Number.isInteger(limit) || limit < 1) throw new Error("--limit 必须是正整数");
  const arg = positionals[1] ?? "";
  if (command === "hot") console.log(formatHot(await fetchAihot("/hot-topics")));
  else if (command === "daily") console.log(formatDaily(await fetchAihot("/dailies/latest")));
  else {
    const path = {
      today: "/items?mode=selected&window=24h",
      week: "/items?mode=selected&window=7d&limit=10",
      search: `/items?mode=selected&q=${encodeURIComponent(arg)}&window=7d&limit=10`,
      category: `/items?mode=selected&category=${encodeURIComponent(arg || "model")}&window=24h`,
      all: `/items?mode=all&window=24h&limit=${limit}`,
    }[command];
    console.log(formatItems(await fetchAihot(path), limit));
  }
  return 0;
}

if (process.argv[1] && fileURLToPath(import.meta.url) === process.argv[1]) {
  main().then(
    (code) => { process.exitCode = code; },
    (error) => {
      console.error(`[错误] ${error instanceof Error ? error.message : String(error)}`);
      process.exitCode = 1;
    },
  );
}
